sunpy/map/mapbase.py

```diff
@@ -1482,6 +1483,10 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
         pad_x = int(np.max((diff[1], 0)))
         pad_y = int(np.max((diff[0], 0)))
 
+        if issubclass(self.data.dtype.type, numbers.Integral) and isinstance((missing), float):
+            warnings.warn("Integer map data is incompatibe with specified missing value",
```

Comment:
```suggestion
            warnings.warn("Integer map data is incompatible with specified missing value",
```
If the value was a float of 1.0, that is fine as an int, even if it was 1.5, it gets rounded by you can cast the value.

I wonder if this is only for nans.

Reply:
any `float` value except for `.0` will result in the same issue, just checked. So I guess unless the user specifies a `missing=1.0`. Either way the warning should be raised right?

Reply:
I wonder if raising a different message such as "irregular behavior might occur" is more appropriate?

Reply:
I am not sure I want to raise a warning when a float is also a int without a loss of information.
Otherwise yeah I think for 1.1, it should raise the warning.